imscjs-demo/js/index.js

```diff
@@ -1,59 +1,48 @@
-
-function initTTML() {
-  var renderDiv = document.getElementById("render-div");
-  var myVideo = document.getElementById("imscVideo");
-  //Track object generated by the first <track> child element
-  var myTrack = myVideo.textTracks[0];
-  //Src is only available from the <track> DOM element, not the obj
-  var ttmlUrl = myVideo.getElementsByTagName("track")[0].src;
-  //The track is disabled by default, 'hidden' does not display VTT Text
-  //but fires the events
+async function initTTML() {
+  const renderDiv = document.getElementById("render-div");
+  const myVideo = document.getElementById("imscVideo");
+  // Track object generated by the first <track> child element
+  const myTrack = myVideo.textTracks[0];
+  // The source is only available from the <track> DOM element, not the object
+  const ttmlUrl = myVideo.getElementsByTagName("track")[0].src;
+  // The track is disabled by default, 'hidden' does not display VTT Text
+  // but fires the events
   myTrack.mode = "hidden";
 
-  var client = new XMLHttpRequest();
-
-  client.open('GET', ttmlUrl);
-  client.onreadystatechange = function () {
-    initTrack(client.responseText);
-  }
-  client.send();
-
+  const response = await fetch(ttmlUrl);
+  initTrack(await response.text());
 
   function initTrack(text) {
-    var imscDoc = imsc.fromXML(text);
-    var timeEvents = imscDoc.getMediaTimeEvents();
-    //create cue per timed event and render isd
-    for (var i = 0; i < timeEvents.length; i++) {
-      //Edge/IE implement "Generic Cue", other browsers VTTCue
-      var Cue = window.VTTCue || window.TextTrackCue;
+    const imscDoc = imsc.fromXML(text);
+    const timeEvents = imscDoc.getMediaTimeEvents();
+    // Create one cue per timed event and render it
```

Comment:
```suggestion
    // Since `TextTrackCue`/`VTTCue` does not support TTML natively, we create one (empty)
    // cue per event on the timeline of the TTML document. When the cue is triggered 
    // (`enter` event), we render the TTML document corresponding to the cue time, 
    // which is called an Intermediary Synchronic Document (ISD), into `renderDiv`.
```

Reply:
Great comment, thank you!